packages/react-router/src/routeInfo.ts

```diff
@@ -14,7 +14,7 @@ export type RoutesById<TRouteTree extends AnyRoute> = {
 }
 
 export type RouteById<TRouteTree extends AnyRoute, TId> = Extract<
-  Extract<ParseRoute<TRouteTree>, { id: TId }>,
+  RoutesById<TRouteTree>[TId],
```

Comment:
Turns out mapped types are faster than `Extract` on a union ;). Makes sense when you think about it. But it only shows when you start to have larger code bases